DetectorDescription/DDCMS/interface/DDDetector.h

```diff
@@ -2,20 +2,48 @@
 #define DETECTOR_DESCRIPTION_DD_DETECTOR_H
 
 #include "DetectorDescription/DDCMS/interface/DDSpecParRegistry.h"
+#include <string>
 
 namespace dd4hep {
   class Detector;
 }
 
 namespace cms  {
-  struct DDDetector {
-
+  class DDDetector {
+  public:
     using Detector = dd4hep::Detector;
 
-    Detector* description = nullptr;
-    DDVectorsMap vectors;
-    DDPartSelectionMap partsels;
-    DDSpecParRegistry specpars;
+    explicit DDDetector(const std::string&, const std::string&);
+    DDDetector() = delete;
+    
+    ~DDDetector();
+
+    // FIXME: remove the need for it
+    Detector const* description() const {
+      return m_description;
+    }
+    
+    DDVectorsMap const& vectors() const {
+      return m_vectors;
+    }
+
+    DDPartSelectionMap const& partsels() const {
+      return m_partsels;
+    }
+    
+    DDSpecParRegistry const& specpars() const {
+      return m_specpars;
+    }
+
+  private:
+
+    void process(const std::string&);
+    
+    Detector* m_description = nullptr;
```

Comment:
Go ahead and make this
```suggestion
    Detector const* m_description = nullptr;
```

Reply:
it's modified by adding extensions to communicate with the producer.